docs/performance-tuning.asciidoc

```diff
@@ -0,0 +1,81 @@
+ifdef::env-github[]
+NOTE: For the best reading experience,
+please view this documentation at https://www.elastic.co/guide/en/apm/agent/dotnet/current/performance-tuning.html[elastic.co]
+endif::[]
+
+[[performance-tuning]]
+== Performance Tuning
+
+There are many options available to tune agent performance.
+Which option to adjust depends on whether you are optimizing for speed, memory usage, bandwidth or storage.
+
+[float]
+[[performance-tuning-sampling]]
+=== Sampling
+
+The first knob to reach for when tuning the performance of the agent is <<config-transaction-sample-rate>>.
+Adjusting the sample rate controls what percent of requests are traced.
+By default, the sample rate is set at `1.0`, meaning _all_ requests are traced.
+
+The sample rate will impact all four performance categories,
+so simply turning down the sample rate is an easy way to improve performance.
+
+Here's an example of setting the sample rate to 20% using <<configuration-on-asp-net-core>>:
+
+[source,js]
+----
+{
+    "ElasticApm": {
+        "TransactionSampleRate": 0.2
+    }
+}
+----
+
+[float]
+[[performance-tuning-stack-traces]]
+=== Stack Traces
+
+In a complex application,
+a request may produce many spans.
+Capturing a stack trace for every span can result in significant memory usage.
+Stack traces are also captured for every error.
+There are several settings to adjust how stack traces are captured.
+
+[float]
+[[performance-tuning-disable-capturing-stack-traces]]
+==== Disable capturing stack traces
+
+To disable capturing stack traces (for both spans and errors),
+set <<config-stack-trace-limit,`StackTraceLimit`>> to `0`.
```

Comment:
Same as above, is the `StackTraceLimit` needed? There are similar things blow in this file.

Reply:
Please see, my comment above.

Reply:
> I'm not sure I understand what you mean by

I meant that there is the similar pattern in this file. E.g. `<<config-span-frames-min-duration,`SpanFramesMinDuration`>>.`

As discussed above, probably this is fine. Would be nice to hear from @bmorelli25 or @Titch990  if we are ok to use this. I think it's fine.